pkg/crc/machine/bundle/metadata.go

```diff
@@ -244,6 +252,35 @@ func GetBundleNameFromURI(bundleURI string) string {
 	}
 }
 
+// GetBundleInfoFromName Parses the bundle filename and returns a FilenameInfo struct
+func GetBundleInfoFromName(bundleName string) (*FilenameInfo, error) {
+	var filenameInfo FilenameInfo
+	bundleName = GetBundleNameWithoutExtension(bundleName)
+
+	// crc_preset_driver_version_arch_customSuffix.crcbundle
+	bundleNameRegex := regexp.MustCompile(`crc(?:(?:_)([[:alpha:]]+))?_([[:alpha:]]+)_([0-9.]+)_([[:alnum:]]+)(?:(?:_)([0-9]+))?`)
+	filenameParts := bundleNameRegex.FindStringSubmatch(bundleName)
+	if filenameParts == nil {
+		return &filenameInfo, fmt.Errorf("bundle filename is in unrecognized format")
+	}
+
+	if filenameParts[1] == "" {
+		filenameInfo.Preset = crcPreset.OpenShift
+	} else {
+		parsedPreset, err := crcPreset.ParsePresetE(filenameParts[1])
+		if err != nil {
+			return &filenameInfo, err
+		}
+		filenameInfo.Preset = parsedPreset
+	}
+	filenameInfo.Driver = filenameParts[2]
+	filenameInfo.Version = filenameParts[3]
+	filenameInfo.Arch = filenameParts[4]
+	filenameInfo.CustomBundleSuffix = filenameParts[5]
+
+	return &filenameInfo, nil
+}
+
```

Comment:
We should have unit test for this function which takes all the valid bundle name and provide info what it suppose to do and some dummy one to check the error.

Reply:
I added a unit test for both valid and invalid bundle filenames.